dlp/api/Snippets/SendDataToTheHybridJobTrigger.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+// [START dlp_inspect_send_data_to_hybrid_job_trigger]
+
+using System;
+using System.Threading;
+using Google.Cloud.Dlp.V2;
+
+public class SendDataToTheHybridJobTrigger
+{
+    public static DlpJob SendToHybridJobTrigger(
+       string projectId,
+       string jobId,
+       string jobPath,
+       string text = null)
+    {
+        // Instantiate the dlp client.
+        var dlp = DlpServiceClient.Create();
+
+        // Construct the hybrid finding details which will be used as metadata with the content.
+        // Refer to this for more information: https://cloud.google.com/dlp/docs/reference/rest/v2/Container
+        var findingDetails = new HybridFindingDetails
+        {
+            ContainerDetails = new Container
+            {
+                FullPath = "10.0.0.2:logs1:aap1",
+                RelativePath = "app1",
+                RootPath = "10.0.0.2:logs1",
+                Type = "System Logs"
+            }
+        };
+
+        // Construct the hybrid content item using the finding details and text to be inspected.
+        var hybridContentItem = new HybridContentItem
+        {
+            Item = new ContentItem { Value = text ?? "My email is ariel@example.org and name is Ariel." },
+            FindingDetails = findingDetails
+        };
+
+        // Activate the Job Trigger.
+        var activate = new ActivateJobTriggerRequest
+        {
+            JobTriggerName = new JobTriggerName(projectId, jobId),
+            Name = jobPath
+        };
+
+        // Call the API to activate the trigger.
+        DlpJob triggerJob = dlp.ActivateJobTrigger(activate);
+
+        // Construct the request using hybrid content item.
+        var request = new HybridInspectJobTriggerRequest
+        {
+            HybridItem = hybridContentItem,
+            JobTriggerName = new JobTriggerName(projectId, jobId),
+            Name = jobPath,
+        };
+
+        // Call the API.
+        HybridInspectResponse response = dlp.HybridInspectJobTrigger(request);
+
+        // Check the activated job status to fetch the inspect details.
```

Comment:
Wouldn't this be demonstrated as well without the iteration?

I think the sample as it is its given the impression that for the status etc, to be valid, you need the job to have processes some bytes? If you just get the job once and print it status etc. wouldn't that be demonstration enough of how to use the library?

Reply:
The response of  `dlp.HybridInspectJobTrigger(request);` is `{ }`. And if we just get the job and does not wait for the job to get completed; It will not show the different findings that job has found. Without waiting, the job state would be `Pending` or `Running` which will not help much while asserting in the test cases. That is why we have to wait (iterations) for the job to get completed

Reply:
Then check the job is in a done state, instead of checking if it has processed some bytes. It's clearer.

Reply:
As for waiting on the sample so that you can get your tests to pass, that shoouldn't be necessary. You can wait in the test itself.

If you want to show the ser how to wait for the job to be completed, then that's fine, but do not add code to a sample for the sake of the tests.

Reply:
The job inside hybrid job trigger does not contain the status `Done`. Whenever it finishes the job the status will be `Active`

Reply:
Its not for the sake of tests, we are mocking the response anyways. To provide more information about the inspection details, we have waited for the job to process the bytes and showing the results.